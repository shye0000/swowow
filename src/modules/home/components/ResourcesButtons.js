import React from 'react';
import Button from 'antd/lib/button';
import Animate from 'rc-animate';

const supportedResources = ['planets'];

const getResourceHref = (resource) => {
	switch (resource) {
		case 'planets': {
			return '/#/planets';
		}
	}
};

const ResourcesButtons = ({root}) => {
	return <div className="resources-buttons">
		<Animate
			transitionName="fade"
			transitionAppear
		>
			{Object.keys(root).map((resource) => {
				return <Button
					disabled={!supportedResources.includes(resource)}
					key={resource}
					href={getResourceHref(resource)}
				>{resource}</Button>;
			})}
		</Animate>
	</div>;
};

export default ResourcesButtons;