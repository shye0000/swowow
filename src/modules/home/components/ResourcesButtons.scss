.resources-buttons {
  button, a {
    border-style: dashed;
    border-width: 2px;
    &[disabled] {
      color: white;
      cursor: not-allowed;
      opacity: 0.5;
    }
  }
}