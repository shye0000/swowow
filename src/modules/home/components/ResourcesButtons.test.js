import React from 'react';
import ResourcesButtons from './ResourcesButtons';
import { shallow } from 'enzyme';

const setup = () => {
	// simulate props
	const props = {
		root: {
			planets: 'url_planets',
			people: 'url_people',
		}
	};
	const wrapper = shallow(<ResourcesButtons {...props} />);
	return {
		props,
		wrapper
	};
};

describe('ResourcesButtons', () => {

	const { wrapper } = setup();

	it('render correctly', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('render with 2 buttons', () => {
		expect(wrapper.find('Button')).toHaveLength(2);
	});

});