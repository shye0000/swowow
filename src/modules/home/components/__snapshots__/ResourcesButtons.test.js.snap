// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourcesButtons render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResourcesButtons
    root={
      Object {
        "people": "url_people",
        "planets": "url_planets",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Button
          disabled={false}
          ghost={true}
          href="#/planets"
          loading={false}
          prefixCls="ant-btn"
          type="dashed"
        >
          planets
        </Button>,
        <Button
          disabled={true}
          ghost={true}
          href="#/"
          loading={false}
          prefixCls="ant-btn"
          type="dashed"
        >
          people
        </Button>,
      ],
      "className": "resources-buttons",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "planets",
        "nodeType": "class",
        "props": Object {
          "children": "planets",
          "disabled": false,
          "ghost": true,
          "href": "#/planets",
          "loading": false,
          "prefixCls": "ant-btn",
          "type": "dashed",
        },
        "ref": null,
        "rendered": "planets",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "people",
        "nodeType": "class",
        "props": Object {
          "children": "people",
          "disabled": true,
          "ghost": true,
          "href": "#/",
          "loading": false,
          "prefixCls": "ant-btn",
          "type": "dashed",
        },
        "ref": null,
        "rendered": "people",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Button
            disabled={false}
            ghost={true}
            href="#/planets"
            loading={false}
            prefixCls="ant-btn"
            type="dashed"
          >
            planets
          </Button>,
          <Button
            disabled={true}
            ghost={true}
            href="#/"
            loading={false}
            prefixCls="ant-btn"
            type="dashed"
          >
            people
          </Button>,
        ],
        "className": "resources-buttons",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "planets",
          "nodeType": "class",
          "props": Object {
            "children": "planets",
            "disabled": false,
            "ghost": true,
            "href": "#/planets",
            "loading": false,
            "prefixCls": "ant-btn",
            "type": "dashed",
          },
          "ref": null,
          "rendered": "planets",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "people",
          "nodeType": "class",
          "props": Object {
            "children": "people",
            "disabled": true,
            "ghost": true,
            "href": "#/",
            "loading": false,
            "prefixCls": "ant-btn",
            "type": "dashed",
          },
          "ref": null,
          "rendered": "people",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
